notifications.bs

```diff
@@ -1002,69 +1007,90 @@ partial interface ServiceWorkerGlobalScope {
 };
 </pre>
 
-<p>The
-<dfn method for=ServiceWorkerRegistration><code>showNotification(title, options)</code></dfn>
-method, when invoked, must run these steps:
+<p>The <dfn method for=ServiceWorkerRegistration><code>showNotification(title, options)</code></dfn>
+method steps are:
 
 <ol>
-  <li><p>Let <var>promise</var> be a new promise.
+ <li><p>Let <var>global</var> be <a>this</a>'s <a>relevant global object</a>
 
-  <li><p>If the <a>context object</a>'s <a for="service worker registration">active worker</a> is
-  null, then reject <var>promise</var> with a <code>TypeError</code> exception and return
-  <var>promise</var>.
+ <li><p>Let <var>promise</var> be <a>a new promise</a> in <var>global</var>'s
+ <a for="global object">Realm</a>.
 
-  <li><p>Let <var>serviceWorkerRegistration</var> be the <a>context object</a>.
+ <li><p>If <a>this</a>'s <a for="service worker registration">active worker</a> is null, then
+ <a for=/>reject</a> <var>promise</var> with a {{TypeError}} and return <var>promise</var>.
+ <!-- We might have to throw directly here and below, see
+      https://github.com/web-platform-tests/wpt/pull/24601 -->
 
-  <li><p>Let <var>notification</var> be the result of <a>creating a notification</a> given
-  <var>title</var>, <var>options</var>, and <var>serviceWorkerRegistration</var>. If this threw an
-  exception, reject <var>promise</var> with that exception and return <var>promise</var>.
+ <li><p>Let <var>serviceWorkerRegistration</var> be <a>this</a>.
 
-  <li><p>Run these steps <a>in parallel</a>:
+ <li><p>Let <var>notification</var> be the result of <a>creating a notification</a> given
+ <var>title</var>, <var>options</var>, and <var>serviceWorkerRegistration</var>. If this threw an
+ exception, then <a for=/>reject</a> <var>promise</var> with that exception and return
+ <var>promise</var>.
+
+ <li>
+  <p>Run these steps <a>in parallel</a>:
 
   <ol>
-    <li><p>If <a>permission</a> for <var>notification</var>'s <a for=notification>origin</a> is not
-    "<code>granted</code>", then reject <var>promise</var> with a <code>TypeError</code> exception,
-    and abort these steps.
+   <li><p>If <a>permission</a> for <var>notification</var>'s <a for=notification>origin</a> is not
+   "<code>granted</code>", then <a>queue a global task</a> on the
+   <a>DOM manipulation task source</a> given <var>global</var> to <a for=/>reject</a>
+   <var>promise</var> with a {{TypeError}}, and abort these steps.
 
-    <li><p>Run the <a>fetch steps</a> for <var>notification</var>.
+   <li><p>Run the <a>fetch steps</a> for <var>notification</var>.
```

Comment:
Probably a separate issue but the fetching also needs some modernization on top of the various processResponse/useParallelQueue stuff..